esm/html/element.js

```diff
@@ -31,8 +31,15 @@ export class HTMLElement extends Element {
 
   constructor(ownerDocument = null, localName = '') {
     super(ownerDocument, localName);
+    const {constructor: Class, [END]: end} = this;
+    if (upgradingElements.has(Class)) {
+      const {element, values } = upgradingElements.get(Class);
+      upgradingElements.delete(Class);
```

Comment:
could you please reason this?

Reply:
let me try again ... could you please reason this?

```js
upgradingElements.delete(Class);
```

why would you drop the class? what about further "same class" elements?

Reply:
I'm using the `upgradingElements` `WeakMap` as context object between the `HTMLElement` and the `CustomElementRegistry`, as its imho the best way to pass objects around between modules and not preventing garbage collection. In contrast in the webcomponents polyfill they used a module scoped variable to track the element that is currently beeing upgraded. 

`WeakMaps` require an object as key, so it felt natural to me to use the `Class` of the custom element, because the `Class` is also known in the `HTMLElement` constructor. As the `value` of the map entry I needed to use the currently upgrading `element` that needs to be returned inside the constructor. 

The value of the `upgradingElements` entry with the class key is always a single particular instance of an `HTMLElement` in the DOM. If I wouldn't delete the entry from the map, an instantiation like `new CustomElement()` would reference the last upgrading element instance related to that custom element class, which leads to very nasty bugs.

The assumptions here are that upgrading a single element happens synchronously and every custom element is a child of `HTMLElement`.